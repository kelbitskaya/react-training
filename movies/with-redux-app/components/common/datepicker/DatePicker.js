import React from "react";
import { useField, useFormikContext } from "formik";
import PropTypes from 'prop-types';
import DatePicker from "react-datepicker";
import 'react-datepicker/dist/react-datepicker.css';

export const DatePickerField = ({ ...props }) => {
  const { setFieldValue } = useFormikContext();
  const [field] = useField(props);
  return (
    <DatePicker
      /* eslint-disable react/jsx-props-no-spreading */
      {...field}
      {...props}
      selected={(field.value && new Date(field.value)) || null}
      onChange={val => {
        setFieldValue(field.name, val);
      }}
      className={props.className}
    />
  );
};

DatePickerField.propTypes = {
  className: PropTypes.string.isRequired,
};

export default DatePickerField;
