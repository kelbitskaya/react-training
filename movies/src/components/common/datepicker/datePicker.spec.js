import React from 'react';
import rerender from 'react-test-renderer';
import DatePicker from './DatePicker';
import { Formik, Field } from "formik";

const props = {
  className: ''
};

describe('DatePicker component', () => {
  let onChange;
  let setFieldValue;

  beforeEach(()=>{
    onChange = jest.fn();
    setFieldValue = jest.fn();
  });

  test('DatePicker layout', () => {
    const component = rerender.create(
      <Formik>
        <Field>
          <DatePicker {...props} onChange={onChange}/>
        </Field>
      </Formik>
      );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
