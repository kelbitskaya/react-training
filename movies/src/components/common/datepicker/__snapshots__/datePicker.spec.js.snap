// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker component DatePicker layout 1`] = `
<input
  onBlur={[Function]}
  onChange={[Function]}
>
  <div
    className="react-datepicker-wrapper"
  >
    <div
      className="react-datepicker__input-container"
    >
      <input
        className=""
        disabled={false}
        name="test"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        readOnly={false}
        type="text"
        value=""
      />
    </div>
  </div>
</input>
`;
